### ISSUE TITLE
error: subprocess-exited-with-error

### ISSUE THREAD

**Issue by Lochan9** [githubharald/CTCWordBeamSearch#71]
When I am trying to install "pip install ." I am getting this error 
help me understand this
![image](https://user-images.githubusercontent.com/63242350/232086208-1b227919-0310-4232-b553-adcd167e0ebb.png)

**Comment by githubharald** [comment]
The package contains C++ code, so you need to install a C++ compiler on your computer.

**Comment by JOHNATHANHOMER00** [comment]
Hi I received the same issue, once I installed the latest version of visual studio I received another error I have been trying to troubleshoot the error but when I fix one another pops up, not sure what to do. If there is a specific version of Visual Studio you recommend please let me know. Thanks.

For reference :
![CMD](https://user-images.githubusercontent.com/61136223/232491931-e8dc01ca-9760-4ca7-9fae-af9d214049db.PNG)

![vs](https://user-images.githubusercontent.com/61136223/232491924-a8fc9e52-b28d-4fcf-9d70-8b4799eebcff.PNG)

![tools](https://user-images.githubusercontent.com/61136223/232491933-27129a15-fae1-41b4-82d9-6b41632ad23c.PNG)

**Comment by filipeas** [comment]
@JOHNATHANHOMER00 I was also having the same problem. I tried a little of everything, but I came to a conclusion that I believe is ideal due to the version of c++ implemented in this project. I did the following:

- First, I installed visual studio build 2019 [here the link](https://winstall.app/apps/Microsoft.VisualStudio.2019.BuildTools).
![image](https://user-images.githubusercontent.com/23065588/234417637-77413179-fa38-47e0-8ec2-bb4cc18211ed.png)
![image](https://user-images.githubusercontent.com/23065588/234417687-dfdbdfbb-adb5-44f7-bf58-7f11a26fdee3.png)

- After I tried it, I noticed that cl.exe is compatible with python 3.8.10. So I used anaconda and created a clean env with that exact python version.

After that, I restarted the computer and run ```pip install .``` inside the project directory and it worked.

![image](https://user-images.githubusercontent.com/23065588/234417844-25e578f4-78d4-4ed0-889a-fc35cb45180b.png)

I hope this helps.
